edb/server/dbview/dbview.pyx

```diff
@@ -148,6 +160,73 @@ cdef class Database:
     def tenant(self):
         return self._index._tenant
 
+    def stop(self):
+        if self._cache_worker_task:
+            self._cache_worker_task.cancel()
+            self._cache_worker_task = None
+
+    async def cache_worker(self):
+        while True:
+            try:
+                await self._cache_worker()
+            except Exception as ex:
+                debug.dump(ex)
+                metrics.background_errors.inc(
+                    1.0, self.tenant._instance_name, "cache_worker"
+                )
+
+    async def _cache_worker(self):
+        added_since_signal = 0
+
+        i = 0
+        while True:
+            # First, handle any evictions
+            keys = []
+            while self._eql_to_compiled.needs_cleanup():
+                query_req, unit_group = self._eql_to_compiled.cleanup_one()
+                if len(unit_group) == 1 and unit_group.cache_state == 1:
+                    keys.append(query_req.get_cache_key())
+                unit_group.cache_state = CacheState.Evicted
+            if keys:
+                await self.tenant.evict_query_cache(self.name, keys)
+
+            # Now, populate the cache
+            query_req, units = await self._cache_queue.get()
```

Comment:
Do you want to drain the queue right here to group DB writes into one query?

Reply:
Basically I'd do an `await queue.get()` followed by a series of `queue.get_nowait()` until they complain

Reply:
Sure. @fantix's existing infrastructure makes it super trivial to do